### PR TITLE
Cleanup: Remove CDUtils

### PR DESCRIPTION
**Pull request by MayImilae**
This is a cleaning followup for #11456, Remove Boot from DVD Backup. CDUtils is a self-described shameless rip of [libcdio](https://www.gnu.org/software/libcdio/), a linux library for accessing drives. It did a lot of heavy lifting for the Boot from DVD Backup feature, despite not being directly connected to it and not giving any errors when I removed Boot from DVD Backup. That and the fact that it only mentions _CDs_, not _DVDs_, is why I missed this in the initial PR. Now it probably does absolutely nothing, and even if it does do something it would be broken without DriveBlob.

This PR is _specifically_ a removal of CDUtils. There are potentially more Boot from DVD Backup bits remaining, but DiscIO and DVDInterface are a little confusing to me and I didn't want to rip out anything that Dolphin needs to boot ISOs. So there is possibly more cleanup left to do, but that will be for another PR.

Please review!

-----

Also... if anyone decides to reimplement Boot from DVD Backup, please rebuild it into something better rather than restoring this mess. The way cdutils+driveblob worked and interacted together (or more specifically, the complete lack thereof) was very awkward and weird. It would be so much better if all physical drive shenanigans were together in one file, and it would be much cleaner too.

### CODE REVIEW

**Comment by MayImilae** [comment]
@JosJuice I especially would like for you to review this.

File: Source/Core/DolphinLib.props [hunk 1]
```diff
@@ -20,7 +20,6 @@
     <ClInclude Include="Common\BitSet.h" />
     <ClInclude Include="Common\BitUtils.h" />
     <ClInclude Include="Common\BlockingLoop.h" />
-    <ClInclude Include="Common\CDUtils.h" />
```
Comment (Simonx22) [inline Source/Core/DolphinLib.props:23]: Seems like you forgot to remove `CDUtils.cpp`. Removing it should fix the Windows build. https://github.com/dolphin-emu/dolphin/blob/0385a414a177eccc0d204ed461736e26d1316bc8/Source/Core/DolphinLib.props#L722

**Review by Simonx22 — CHANGES_REQUESTED** [review]

File: Source/Core/DolphinLib.props (hunk 1 above)
Comment (MayImilae) [reply Source/Core/DolphinLib.props:23]: Done!

**Comment by MayImilae** [comment]
Tested on Kubuntu, macOS 12, and Windows 11. Games still boot!

File: Source/Core/Core/Boot/Boot.cpp [hunk 2]
```diff
@@ -273,14 +271,6 @@ std::unique_ptr<BootParameters> BootParameters::GenerateFromFile(std::vector<std
...
 
-    if (is_drive)
-    {
-      PanicAlertFmtT("Could not read \"{0}\". "
-                     "There is no disc in the drive or it is not a GameCube/Wii backup. "
-                     "Please note that Dolphin cannot play games directly from the original "
-                     "GameCube and Wii discs.",
-                     path);
-    }
     else
```
Comment (BhaaLseN) [inline Source/Core/Core/Boot/Boot.cpp:274]: You should probably get rid of this `else` and only leave its contents. GitHub won't let me add a working suggestion tho; so remove the `else` and the curlies, but leave the `PanicAlertFmtT`.

Comment (MayImilae) [reply Source/Core/Core/Boot/Boot.cpp]: Done!

**Review by BhaaLseN — APPROVED** [review]

**Comment by JosJuice** [comment]
Putting the question of whether the disc drive feature should be removed aside: The code change LGTM, but I have some feedback regarding the commit message.

> This is a cleaning followup for https://github.com/dolphin-emu/dolphin/pull/11456, Remove Boot from DVD Backup. CDUtils is a self-described shameless ripoff of cdio, a linux library for accessing drives. It did a lot of the heavy lifting for the Boot from DVD Backup feature, despite not being connected to it in any way and not giving any errors when removing Boot from DVD Backup. That and the fact that DVD is not anywhere within it is why I missed this in the initial PR.

The writing here makes sense, but there are no line breaks in the text. For better or worse, in commit messages you're supposed to place line breaks manually.

> CDUtil has fingers in lots of things. Despite the many files changed here, this is a fairly conservative removal. At this level it gets really hard to distinguish things necessary for actual physical drives and things needed for emulated drives, and I didn’t want to rip out anything that Dolphin needs to boot ISOs. There is likely still some DVD bits I missed though, though this is the last one I’m aware of.

The code change you made seems like a perfectly standard removal of something that isn't being used anymore. You removed the CDUtils files, all includes of them, and all uses of functions defined in them. I don't think it makes sense to say that CDUtils had fingers in lots of things or that this is a conservative removal.

> This absolutely needs thorough review.

Personally I put things like this in the PR description but not the commit message. After all, by the time the PR is merged, the commit will have been properly reviewed and thus there's no reason to have this text in the commit anymore. But maybe that's just me.

**Comment by MayImilae** [comment]
Ahh I was not aware there was any sort of commit message etiquette. In fact i wasn't really aware there was a difference between commit message and PR description!

Do you have any more tips / guidance regarding this? This is a new concept for me and I'd like to know more!

**Comment by JosJuice** [comment]
Would prefer if the commit message is wrapped at 72 characters instead of at the end of each sentence, but if someone else wants to merge this, I'm not going to block it.

**Comment by MayImilae** [comment]
The commit message is now under 72 characters per line. Everyone writing to Dolphin on their IBM 3270 will be pleased.

Also rebased so there are no longer any spurious fifoci errors.

**Pull request merged**